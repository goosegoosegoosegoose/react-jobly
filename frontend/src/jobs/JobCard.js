import React, { useContext, useEffect, useState } from "react";
import UserContext from "../auth/userContext";

const JobCard = ({title, salary, equity, apply, id}) => {
  const currentUser = useContext(UserContext);
  const [user, setUser] = useState(currentUser);
  const exists = user.applications.includes(id);
  // currentUser sometimes is defined, sometimes is not

  useEffect(() => {
    setUser(currentUser);
  }, [currentUser])
  
  return (
    <div>
      <h6>{title}</h6>
      <small>Salary: {salary}</small><br />
      <small>Equity: {equity}</small><br />
      {exists ? <button disabled>Applied</button> : <button id={id} onClick={apply}>Apply</button>}
    </div>
  )
}

export default JobCard;