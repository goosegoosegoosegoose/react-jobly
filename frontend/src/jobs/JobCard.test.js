import { render } from '@testing-library/react';
import JobCard from './JobCard'  

test('smoke test', () => {
  render(      
    <JobCard />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<JobCard />);
  expect(asFragment()).toMatchSnapshot();
});