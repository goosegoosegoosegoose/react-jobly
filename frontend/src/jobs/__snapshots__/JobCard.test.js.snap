// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <h6 />
    <small>
      Salary: 
    </small>
    <br />
    <small>
      Equity: 
    </small>
  </div>
</DocumentFragment>
`;
